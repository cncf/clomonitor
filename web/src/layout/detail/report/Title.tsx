interface Props {
  title: string;
  icon: JSX.Element;
  className?: string;
  anchor?: JSX.Element;
}

const Title = (props: Props) => {
  return (
    <div className={`d-flex flex-row align-items-center ${props.className}`}>
      {props.icon}
      <div className="ms-2 fw-bold">{props.title}</div>
      {props.anchor}
    </div>
  );
};

export default Title;
