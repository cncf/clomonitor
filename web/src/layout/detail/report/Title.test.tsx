import { render, screen } from '@testing-library/react';

import Title from './Title';

const defaultProps = {
  title: 'title',
  icon: <>icon</>,
};

describe('Title', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Title {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders option', () => {
      render(<Title {...defaultProps} />);

      expect(screen.getByText('title')).toBeInTheDocument();
      expect(screen.getByText('icon')).toBeInTheDocument();
    });
  });
});
