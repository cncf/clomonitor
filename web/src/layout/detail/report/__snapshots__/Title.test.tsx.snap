// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-row align-items-center undefined"
  >
    icon
    <div
      class="ms-2 fw-bold"
    >
      title
    </div>
  </div>
</DocumentFragment>
`;
